pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -353,3 +515,87 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 	log.Debugf("cluster flavor %q not recognised. Defaulting to false", c.clusterFlavor)
 	return false
 }
+
+// IsFakeAttachAllowed checks if the volume is eligible to be fake attached and returns a bool value
+func (c *K8sOrchestrator) IsFakeAttachAllowed(ctx context.Context, volumeID string, volumeManager cnsvolume.Manager) (bool, error) {
+	log := logger.GetLogger(ctx)
+	//Check pvc annotations
+	pvcAnn, err := c.getPVCAnnotations(ctx, volumeID)
+	if err != nil {
+		log.Errorf("failed to get pvc annotations")
+		return false, err
+	}
+
+	if val, found := pvcAnn[common.AnnIgnoreInaccessiblePV]; found && val == "yes" {
+		log.Debugf("Found %s annotation on pvc for volume %s set to yes", common.AnnIgnoreInaccessiblePV, volumeID)
+		//Check if volume is inaccessible
+		log.Debugf("Checking volume health on CNS volume: %s", volumeID)
+		vol, err := common.QueryVolumeByID(ctx, volumeManager, volumeID)
+		if err != nil {
+			log.Errorf("failed to query CNS volume")
+			return false, err
+		}
+
+		if vol.HealthStatus != string(pbmtypes.PbmHealthStatusForEntityUnknown) {
+			volHealthStatus, err := common.ConvertVolumeHealthStatus(vol.HealthStatus)
+			if err != nil {
+				log.Errorf("invalid health status: %s for volume: %s", vol.HealthStatus, vol.VolumeId.Id)
+				return false, err
+			}
+			log.Debugf("CNS volume health is: %s", volHealthStatus)
+
+			// If volume is inaccessible, it can be fake attached.
+			if volHealthStatus == common.VolHealthStatusInaccessible {
+				log.Infof("Volume: %s is eligible to be fake attached", volumeID)
+				return true, nil
+			}
+		}
+		// For all other cases, return false
+		return false, nil
+	}
+	// Annotation is not found or not set to true, return false
+	log.Debugf("Annotation %s not found or not set to true on pvc for volume %s", common.AnnIgnoreInaccessiblePV, volumeID)
+	return false, nil
+}
+
+// MarkFakeAttached updates the pvc corresponding to volume to have a fake attach annotation.
+func (c *K8sOrchestrator) MarkFakeAttached(ctx context.Context, volumeID string) error {
+	log := logger.GetLogger(ctx)
+	annotations := make(map[string]string)
+	annotations[common.AnnVolumeHealth] = common.VolHealthStatusInaccessible
+	annotations[common.AnnFakeAttached] = "yes"
+
+	// Update annotations.
+	// Along with updating fake attach annotation on pvc, also update the volume health on pvc
+	// as inaccessible, as that's one of the conditions for volume to be fake attached.
+	if err := c.updatePVCAnnotations(ctx, volumeID, annotations); err != nil {
+		log.Errorf("failed to update pvc annotations for volume: %s. Error:%+v", volumeID, err)
```

Comment:
Could you make the log message a bit more specific something like - "failed to mark fake attached annotation on the pvc %s and volume %s"